src/RosNode.test.ts

```diff
@@ -30,7 +30,7 @@ describe("RosNode", () => {
 
     const nodeB = new RosNode({
       name: "/nodeB",
-      hostname: "localhost",
+      hostname: "0.0.0.0",
```

Comment:
same question here

Reply:
To fix an error I was getting on that test @aneuwald-ctw found out that changing the hostname to 0.0.0.0 instead would work because usually, as I understood, localhost points to a specific interface locally and 0.0.0.0 points to various interfaces, one of them being the local one. Perhaps because of some internal config that made localhost point to somewhere else. 
I dont know if this fully answers the question and I'm still a bit unsure myself